docs/reference/faq/index.md

```diff
@@ -0,0 +1,429 @@
+failure), Firehose will retry sending the messages for a fixed number of attempts depending on whether a retryable
+exception was thrown. If the messages still failed to get published after retries, Firehose will push these messages to
+the DLQ topic if DLQ is enabled via configs. If not enabled, it will drop the messages.
+
+### JDBC Sink
+
+#### What are some of the use cases of this sink?
+
+This sink can be used for cases when one would want to run customised and frequently changing queries on the streaming
+data, for joining with other data to do complex analytics or for doing transactional processing.
+
+#### What monitoring metrics are available for this sink?
+
+Along with the generic Firehose metrics, this sink also exposes metrics to monitor the following:-
+
+1. Connection pool size for the JDBC connection
+2. Total active and idle connections in the pool
+
+#### Do we need to create table/schema before pushing data via JDBC sink ?
+
+Yes. The table needs to created before Firehose can stream data to it. The table name can be specified via the config **
+SINK_JDBC_TABLE_NAME**.
+
+#### Any restriction of version of supported database?
+
+Firehose JDBC sink works with PostgresSQL ver. 9 and above.
+
+#### How messages get mapped to the queries?
+
+Firehose maps an incoming Kafka message to a query based on 2 configs: **INPUT_SCHEMA_PROTO_TO_COLUMN_MAPPING** and **
+SINK_JDBC_UNIQUE_KEYS**. For example, consider a proto as follows:-
+
+    message Driver {
+        int32 driver_id = 1;
+        PersonName driver_name = 2;
+    }
+    
+    message PersonName {
+        string fname = 1;
+        string lname = 2;
+    }
+
+Now consider that the Driver table in the database has 3 columns: `id`, `first_name` and `last_name`. Accordingly, one
+can set the **INPUT_SCHEMA_PROTO_TO_COLUMN_MAPPING** config as follows:-
+
+    {""1"":""id"", ""2"":""{\""1\"":\""first_name\"", \""2\"":\""last_name\""}""}
+
+This will output a row in the Driver table as follows:-
+
+| id   | first_name | last_name |
+|------|------------|-----------|
+| 1234 | Chuck      | Norris    |
+
+Unique keys as per the table schema can be configured by setting the config **SINK_JDBC_UNIQUE_KEYS**. For
+example: `order_number,driver_id`. When trying to push a message from Kafka to the database table, Firehose will
+construct a query which first tries to do an INSERT. If there is a conflict, such as a unique key constraint violation,
+the query tries to do an update of all the non-unique columns of the row with appropriate values from the message. If
+still there is a conflict, the query won't do anything.
+
+#### How data types are handled?
+
+Firehose can insert/update rows into the DB table with columns of data type string only. For a field of type as a nested
+Protobuf message, Firehose will serialise this message into a JSON string. While serialising, it will trim insignificant
```

Comment:
Can we make it as a sublist ?

1. NestedMessages
2. Collections
3. Timestamp

Reply:
Done as part of bc0843405a62f4f4550bdccb2563a5ffed137bf7